src/charm.py

```diff
@@ -64,6 +71,78 @@ def __init__(self, *args):
         self.framework.observe(self.on.update_status, self._on_update_status)
         self._storage_path = self.meta.storages["pgdata"].location
 
+    def _on_old_db_relation_changed(self, event: RelationChangedEvent) -> None:
```

Comment:
It was suggested on another PR that relations could be their own modules sitting in a "relations" subdirectory, in the same way as our other `Relation` objects. I've found this cleans up my charm code, and allows space to independently document the relation. You can also keep test files clean by following the same structure - one file per relation, contained in a "relations" subdir.